moveit_ros/warehouse/CMakeLists.txt

```diff
@@ -16,9 +16,72 @@ find_package(tf2_ros REQUIRED)
 # Finds Boost Components
 include(ConfigExtras.cmake)
 
-include_directories(warehouse/include)
+include_directories(include)
 
-add_subdirectory(warehouse)
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+  Boost
+  moveit_core
+  rclcpp
+  warehouse_ros
+  moveit_ros_planning
+  tf2_eigen
+  tf2_ros
+)
+
+# Libraries
+add_library(moveit_warehouse SHARED
+  src/moveit_message_storage.cpp
+  src/planning_scene_storage.cpp
+  src/planning_scene_world_storage.cpp
+  src/constraints_storage.cpp
+  src/trajectory_constraints_storage.cpp
+  src/state_storage.cpp
+  src/warehouse_connector.cpp
+)
+include(GenerateExportHeader)
+generate_export_header(moveit_warehouse)
+target_include_directories(moveit_warehouse PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
+set_target_properties(moveit_warehouse PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
+ament_target_dependencies(moveit_warehouse ${THIS_PACKAGE_INCLUDE_DEPENDS})
+
+
+# Executables
+add_executable(moveit_warehouse_broadcast src/broadcast.cpp)
+ament_target_dependencies(moveit_warehouse_broadcast ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_warehouse_broadcast moveit_warehouse)
+
+add_executable(moveit_save_to_warehouse src/save_to_warehouse.cpp)
+ament_target_dependencies(moveit_save_to_warehouse ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_save_to_warehouse moveit_warehouse)
+
+add_executable(moveit_warehouse_import_from_text src/import_from_text.cpp)
+ament_target_dependencies(moveit_warehouse_import_from_text ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_warehouse_import_from_text moveit_warehouse)
+
+add_executable(moveit_warehouse_save_as_text src/save_as_text.cpp)
+ament_target_dependencies(moveit_warehouse_save_as_text ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_warehouse_save_as_text moveit_warehouse)
+
+add_executable(moveit_init_demo_warehouse src/initialize_demo_db.cpp)
+ament_target_dependencies(moveit_init_demo_warehouse ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_init_demo_warehouse moveit_warehouse)
+
+add_executable(moveit_warehouse_services src/warehouse_services.cpp)
+ament_target_dependencies(moveit_warehouse_services ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(moveit_warehouse_services moveit_warehouse)
+
+install(
```

Comment:
There is a redundant install starting in line 97. Also, shouldn't `RUNTIME DESTINATION` point to `bin`?

Reply:
Oh thanks! Fixed it, I messed up the rebasing. The `RUNTIME DESTINATION` is right, because in contrast to the libraries the executables need to be installed in `\lib/${PROJECT_NAME}` otherwise they cannot be found by `ros2 run ...`

Reply:
Oh thanks! Fixed it, I messed up the rebasing. The `RUNTIME DESTINATION` is right, because in contrast to the libraries the executables need to be installed in `\lib/${PROJECT_NAME}` otherwise they cannot be found by `ros2 run ...`

Reply:
Oh thanks! Fixed it, I messed up the rebasing. The `RUNTIME DESTINATION` is right, because in contrast to the libraries the executables need to be installed in `\lib/${PROJECT_NAME}` otherwise they cannot be found by `ros2 run ...`